core/io/pck_packer.cpp

```diff
@@ -259,3 +259,8 @@ Error PCKPacker::flush(bool p_verbose) {
 
 	return OK;
 }
+
+PCKPacker::~PCKPacker() {
+	// Do not remove, kept to prevent forward declaration issues, see:
+	// https://github.com/godotengine/godot/pull/80330
+}
```

Comment:
For the record, I haven't reviewed the discussion in depth, but I'm not too fond of:
- Having to remember to add empty destructors, without a clear criterion to know which classes need it.
- Hardcoding the same comment and link to a GitHub issue everywhere.

Reply:
Could enforce destructors universally for safety

Reply:
But the criterion is:
Has a `Ref` to a forward declared class

Reply:
So to clarify the issue here:
A class, which contains a `Ref` to a forward declared class, must on the first time it is invoked, be it directly or indirectly, have a full declaration available, regardless of the access to that specific class

This differs from pointers

So imagine a class `A` references a class `B` which references a class `C` (which is forward declared), then the scope in which `A` is destructed, even if it does *no* interaction *what so ever* with `C` (all that is done internally in `B`), must contain the full declaration of `C`, if it is the first scope which destructs `C`

Reply:
I was looking at some code and realized that we may already have these empty destructors that somehow "fix things", just without any comment or explanation. Throughout the codebase we have empty constructors and destructors present for no apparent reason, probably just as a stylistic choice by some C++ developers. Given what this PR does, I think some of those may accidentally be preventing similar bugs without us even realizing.

So, in my opinion we should definitely ensure this happens in a more organized and obvious manner.